automation/test.sh

```diff
@@ -124,7 +124,14 @@ while [ "$(kubectl get pods -o'custom-columns=status:status.containerStatuses[*]
     sleep 10
 done
 
-kubectl get pods
+# Wait until all pods are running
+while [ -n "$(kubectl get pods --no-headers --all-namespaces | grep -v Running)" ]; do
```

Comment:
"Running" doesn't mean the cluster is operational. For example, haproxy can be running but not in a "ready" state to accept traffic. This will cause the functional tests to fail immediately because they can't contact the cluster. 

Also, you might want to ignore "terminating" phase here.

Reply:
Yes, there is room for improvement, but it's a little better than before.